src/topology/sets/compacts.lean

```diff
@@ -96,17 +96,32 @@ protected def map (f : α → β) (hf : continuous f) (K : compacts α) : compac
 @[simp] lemma coe_map {f : α → β} (hf : continuous f) (s : compacts α) :
   (s.map f hf : set β) = f '' s := rfl
 
+@[simp] lemma map_id (K : compacts α) : K.map id continuous_id = K := compacts.ext $ set.image_id _
+
+lemma map_comp (f : β → γ) (g : α → β) (hf : continuous f) (hg : continuous g) (K : compacts α) :
+  K.map (f ∘ g) (hf.comp hg) = (K.map g hg).map f hf := compacts.ext $ set.image_comp _ _ _
+
 /-- A homeomorphism induces an equivalence on compact sets, by taking the image. -/
-@[simp] protected def equiv (f : α ≃ₜ β) : compacts α ≃ compacts β :=
+@[simps] protected def equiv (f : α ≃ₜ β) : compacts α ≃ compacts β :=
 { to_fun := compacts.map f f.continuous,
   inv_fun := compacts.map _ f.symm.continuous,
   left_inv := λ s, by { ext1, simp only [coe_map, ← image_comp, f.symm_comp_self, image_id] },
   right_inv := λ s, by { ext1, simp only [coe_map, ← image_comp, f.self_comp_symm, image_id] } }
 
+@[simp] lemma equiv_refl : compacts.equiv (homeomorph.refl α) = equiv.refl _ :=
+equiv.ext map_id
+
+lemma equiv_trans (f : α ≃ₜ β) (g : β ≃ₜ γ) :
+  compacts.equiv (f.trans g) = (compacts.equiv f).trans (compacts.equiv g) :=
+equiv.ext $ map_comp _ _ _ _
+
+lemma equiv_symm (f : α ≃ₜ β) : compacts.equiv f.symm = (compacts.equiv f).symm :=
```

Comment:
Could you tag those two as simp? They are the right way per the Zulip discussion.

Reply:
Which discussion?

Reply:
https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/simp.20normal.20form.20for.20.60e.2Eto_equiv.2Esymm.60

Reply:
I don't think that thready is relevant here, unless I missed the relevant message. This isn't about stronger and weaker equivs, this is about functorial equivs and what lemmas to provide for them.

[`equiv.finset_congr`](https://leanprover-community.github.io/mathlib_docs/data/finset/image.html#equiv.finset_congr) seems to have exactly the opposite convention to the one you're suggesting.

I deliberately didn't mark these `simp` because I wanted to avoid bikeshedding which direction should be `simp`!

Reply:
Equiv weakening is the special case of functoriality along a full functor. Having either direction be simp is better than having none.

Reply:
@fpvandoorn, can you tie break here?

Reply:
I'm leaning towards agreeing with Yael here, my gut reaction is that the direction of the `equiv.finset_congr` lemmas are in the wrong direction. That said, I think neither direction is strictly better than the other, and am fine with all three options (one of which is no simp lemma at all)

Reply:
Fwiw I think I agree with the direction of trans, but not of symm. Either way, I'm outvoted here and don't care that much.